.github/workflows/prerelease-tests-on-release-branch.yaml

```
@@ -0,0 +1,77 @@
+# Prerelease Tests
+
+name: Prerelease Tests on Release branch
+on:
+  workflow_dispatch:
+  schedule:
+    # run prerelease tests daily at 5 am UTC
+    - cron: '0 5 * * *'
+    # run coverity tests at 22:00 on every sunday
+    - cron: '0 22 * * SUN'
+
+# The workflow needs the permission to push branches
+permissions:
+  contents: write
+
+jobs:
+  prerelease_test:
+    name: Prerelease test
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Checkout TimescaleDB
+        uses: actions/checkout@v4
+
+      - name: Set RELEASE_BRANCH as per environment
+        run: |
+          OLD_MINOR_VERSION=$(tail -1 version.config | cut -d ' ' -f 3 | cut -d '-' -f 1)
```

If rows are not in the correct order, this will do strange things and potentially extract the wrong version. I would suggest to grep for the correct field so that there are no weird errors happening. This is a perl version that will do what you want.
```suggestion
          OLD_MINOR_VERSION=$(perl -ne 'print $1 if /downgrade_to_version = (\d+\.\d+\.\d+)/' version.config)
```